### PR TITLE
Add common optional dependencies for extensions

### LINKED ISSUE #3435: Consider including hadoop-aws in the lib folder
https://mvnrepository.com/artifact/org.apache.hadoop/hadoop-aws/2.6.0

S3 ingestion does not work without this library in the lib folder, as far as I can tell.

### PR DESCRIPTION
**Pull request by ccaominh**
Fixes #3435.

### Description

Include hadoop-aws and postgres JDBC connector jar to improve out-of-the-box experience for extensions. The mysql JDBC connector jar is not bundled as it is GPL.

<hr>

This PR has:
- [x] been self-reviewed.
- [x] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/licenses.yaml)
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.

### CODE REVIEW

**Comment by jihoonson** [comment]
Would you update https://github.com/apache/druid/blob/master/docs/development/extensions-core/lookups-cached-global.md and https://github.com/apache/druid/blob/master/docs/development/extensions-core/druid-lookups.md as well?

File: pom.xml [hunk 1]
```diff
@@ -1111,6 +1111,11 @@
                     </exclusion>
                 </exclusions>
             </dependency>
+           <dependency>
+               <groupId>org.postgresql</groupId>
+               <artifactId>postgresql</artifactId>
+               <version>42.2.8</version>
```
Comment (jon-wei) [inline pom.xml]: nit: suggest moving the version number definition to the `<properties>` section

File: pom.xml [hunk 2]
```diff
@@ -98,6 +98,7 @@
         <netty4.version>4.1.45.Final</netty4.version>
         <node.version>v10.14.2</node.version>
         <npm.version>6.5.0</npm.version>
+        <postgreql.version>42.2.8</postgreql.version>
```
Comment (jon-wei) [inline pom.xml]: posgreql -> postgresql

Comment (ccaominh) [reply pom.xml]: doh!

**Pull request merged**